src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
I don't disagree, and I strongly agree about avoiding needless indirection/abstraction. I see this case as an exception, it's a way of labeling an important special class of functionality. I'm not concerned about grepping.